tests/table/test_upsert.py

```diff
@@ -366,3 +367,22 @@ def test_upsert_with_identifier_fields(catalog: Catalog) -> None:
 
     assert upd.rows_updated == 1
     assert upd.rows_inserted == 1
+
+
+def test_create_match_filter_single_condition() -> None:
+    """
+    Test create_match_filter with a composite key where the source yields exactly one unique key.
+    Expected: The function returns the single And condition directly.
+    """
+
+    data = [
+        {"order_id": 101, "order_line_id": 1, "extra": "x"},
+        {"order_id": 101, "order_line_id": 1, "extra": "x"},  # duplicate
+    ]
+    schema = pa.schema([pa.field("order_id", pa.int32()), pa.field("order_line_id", pa.int32()), pa.field("extra", pa.string())])
+    table = pa.Table.from_pylist(data, schema=schema)
+    expr = create_match_filter(table, ["order_id", "order_line_id"])
+    expr_str = str(expr)
+
+    assert "And(" in expr_str, f"Expected And condition but got: {expr_str}"
+    assert "Or(" not in expr_str, "Did not expect an Or condition when only one unique key exists"
```

Comment:
@omkenge I think converting it into a string is not the best approach as it doesn't cover the full result. I'd rather see something like:

```suggestion
    assert expr == And(
        EqualTo(term=Reference(name="order_id"), literal=LongLiteral(101)),
        right=EqualTo(term=Reference(name="order_line_id"), literal=LongLiteral(1)),
    )
```